src/lib.rs

```diff
@@ -385,6 +385,9 @@ impl<T: Clone + Integer> Ord for Ratio<T> {
 
         // With equal numerators, the denominators can be inversely compared
         if self.numer == other.numer {
+            if T::is_zero(self.numer()) {
```

Comment:
We should be able to simplify this:

```suggestion
            if self.numer.is_zero() {
```

Reply:
I would go ahead and commit this suggestion, but it seems your branch isn't open to maintainer pushes. If you want to do that, it would be great, or else I can merge and then push that tweak separately.